cdap-master/src/main/java/co/cask/cdap/data/tools/ReplicationStatusTool.java

```diff
@@ -0,0 +1,379 @@
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package co.cask.cdap.data.tools;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.data2.util.TableId;
+import co.cask.cdap.data2.util.hbase.HBaseTableUtil;
+import co.cask.cdap.data2.util.hbase.HBaseTableUtilFactory;
+import co.cask.cdap.data2.util.hbase.ScanBuilder;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.replication.ReplicationConstants;
+import com.google.common.base.Throwables;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.CellUtil;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.client.HTable;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+/**
+ * The Replication Status tool is used to track the status of replication from Master to a Slave Cluster.
+ * The tool reads writeTime and replicateTime for all tables and uses it to determine if all CDAP tables have been
+ * completely replicated to the Slave cluster.
+ * */
+public class ReplicationStatusTool {
+
+  private static CConfiguration cConf = CConfiguration.create();
+  private static Configuration hConf = HBaseConfiguration.create();
+  protected static final Logger LOG = LoggerFactory.getLogger(ReplicationStatusTool.class);
+
+  private static String masterClusterID = null;
+  private static String slaveClusterID = null;
+  private static String localClusterID = null;
+  private static String masterChecksum = null;
+  private static Long masterShutdownTime = 0L;
+  private static String inputStatusFileName = null;
+  private static String outputStatusFileName = null;
+  private static boolean debugDump = false;
+  private static String delimiter = ":";
+
+
+  public static void main (String args[]) throws Exception {
+
+    parseArgs(args);
+    localClusterID = hConf.get(ReplicationConstants.ReplicationStatusTool.CLUSTER_ID);
+    if (localClusterID == null) {
+      System.out.println("Can't read "
```

Comment:
we should either use System.out everything, or use the logger everywhere. The UpgradeTool is a bad example that mixes them. I think using the logger is probably better.